# Process the articels from CoCoMac

In [1]:
# import packages
import pandas as pd
import numpy as np
from nltk import ngrams

In [2]:
# import internal modules
import file_path_management as fpath
import public_library as plib
import parameters as params
import dataframe_columns as df_col

2023-11-18 00:31:47 GM01X680 metapub.config[188129] WARNING NCBI_API_KEY was not set.


In [24]:
df_cocomac = pd.read_csv(fpath.cocomac_literature_list, skiprows=0, sep=',')
df_db = pd.read_csv(fpath.poten_litera_db, header=0, sep='\t')

# print(df_cocomac.columns)
# print(df_db.columns)

# get the list of PMIDs
pmid_list_db = df_db['PMID'].dropna().tolist()
pmid_list_db = [int(i) for i in pmid_list_db]
# print(pmid_list_db)

# Convert 'PMID' to integers, while keeping NaN as is
for ind in df_cocomac.index:
    if df_cocomac.loc[ind, "'PMID'"] == df_cocomac.loc[ind, "'PMID'"]:
        pmid = int(df_cocomac.loc[ind, "'PMID'"].split("'")[1].split("'")[0].strip())
    else:
        pmid = 0
    
    if pmid not in pmid_list_db:
        print(df_cocomac.loc[ind, "'ID'"], df_cocomac.loc[ind, "'ID_Literature'"], df_cocomac.loc[ind, "'Volume'"], df_cocomac.loc[ind, "'PMID'"], df_cocomac.loc[ind, "'Pages'"])

'129' '134' '242' nan '535-560'
'369' '392' '14' '11860458' '1633-1650'
'406' '436' '' '0' '325-343'
'419' '449' '378' '9733085' '562-585'


DOI: 10.1002/cne.902420406 <br>
PMID: 2418080 <br>
The Primate Mediodorsal (MD) Nucleus and Its Projection to the Frontal Lobe <br>
https://pubmed.ncbi.nlm.nih.gov/2418080/ <br>

DOI: 10.1046/j.0953-816x.2001.01789.x <br>
PMID: 11860458 <br>
Organization of inputs from cingulate motor areas to basal ganglia in macaque monkey <br>
https://pubmed.ncbi.nlm.nih.gov/11860458/

DOI: 10.1016/j.tharel.2004.03.001 <br>
PMID: <br>
Circuits through prefrontal cortex, basal ganglia, and ventral anterior nucleus map pathways beyond motor control <br>
https://www.sciencedirect.com/science/article/abs/pii/S1472928804000068

DOI: 10.1002/(SICI)1096-9861(19980928)399:3<384::AID-CNE7>3.0.CO;2-X <br>
PMID: 9733085 <br>
TITLE in CoCoMac: Corticothalamic connections of extrastriate visual areas in rhesus monkeys <br>
TITLE in PubMed: Corticothalamic connections of extrastriate visual areas in rhesus monkeys <br>
https://pubmed.ncbi.nlm.nih.gov/9733085/